scripts/docker/docker.lib.sh

```diff
@@ -150,15 +151,15 @@ function docker-get-image-version-and-pull() {
 
   # Split the image version into two, tag name and digest sha256.
   # E.g. for the given entry "docker/image 1.2.3@sha256:hash" in the
-  # '.tool-versions' file, the following variables will be set:
+  # '.tool-versions' file, the following variables will be set:
   #   version="1.2.3@sha256:hash"
   #   tag="1.2.3"
   #   digest="sha256:hash"
   tag="$(echo "$version" | sed 's/@.*$//')"
   digest="$(echo "$version" | sed 's/^.*@//')"
 
   # Check if the image exists locally already
-  if ! docker images | grep -q "${name}:${tag}"; then
+  if ! docker images | awk '{ print $1 ":" $2 }' | grep "^${name}:${tag}$"; then
```

Comment:
Awk is a [fairly old unix command-line tool](https://en.wikipedia.org/wiki/AWK).  It can be a bit inscrutable, but it's extremely powerful.  What it's doing here is fairly simple, though: it's grabbing the first two fields on each line that `docker images` outputs, and joining them with a `:` character.